src/MongoDB.Driver/Linq/FieldExpressionFlattener.cs

```diff
@@ -86,5 +80,43 @@ protected internal override Expression VisitField(FieldExpression node)
 
             return node;
         }
+
+        // private methods
+        private static string FormatArrayIndex(object index)
+        {
+            if (index is int intIndex)
+            {
+                return GetFormattedIndexOrThrow(intIndex);
+            }
+
+            if (index is long longIndex)
+            {
+                return GetFormattedIndexOrThrow(longIndex);
+            }
+
+            throw new NotSupportedException("Array indexes must be int or long.");
+        }
+
+        private static string GetFormattedIndexOrThrow(long index) 
```

Comment:
We're not "getting" something here, rather "formatting".

Would you be willing to rename this method to:

    FormatArrayIndex

And it can just be an overload with the same name as the existing method. I think it's implied that any method can throw if something goes wrong.

Reply:
Done